src/inference/src/dev/core_impl.cpp

```diff
@@ -304,27 +296,27 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
     auto& config_with_batch = parsed._config;
     // if auto-batching is applicable, the below function will patch the device name and config accordingly:
     apply_auto_batching(model, deviceName, config_with_batch);
-    clean_properties(deviceName, config_with_batch, ov::auto_batch_timeout);
     parsed = parseDeviceNameIntoConfig(deviceName, config_with_batch);
 
     auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<ov::ICompiledModel> res;
-    auto cacheManager =
-        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
-            ._cacheManager;
+    property_manager.merge_property(parsed._config, plugin.get_name());
+    auto cacheManager = property_manager.get_property(ov::cache_manager.name(), plugin.get_name())
+                            .as<std::shared_ptr<ov::ICacheManager>>();
     auto cacheContent = CacheContent{cacheManager};
+    ov::AnyMap _config = property_manager.exclude_property(parsed._config);
     if (cacheManager && device_supports_import_export(plugin)) {
-        cacheContent.blobId = ov::NetworkCompilationContext::compute_hash(
-            model,
-            create_compile_config(plugin, parsed._deviceName, parsed._config));
+        cacheContent.blobId =
+            ov::NetworkCompilationContext::compute_hash(model,
+                                                        create_compile_config(plugin, parsed._deviceName, _config));
         bool loadedFromCache = false;
         auto lock = cacheGuard.get_hash_lock(cacheContent.blobId);
-        res = load_model_from_cache(cacheContent, plugin, parsed._config, context, loadedFromCache);
+        res = load_model_from_cache(cacheContent, plugin, _config, context, loadedFromCache);
         if (!loadedFromCache) {
-            res = compile_model_impl(model, plugin, parsed._config, context, cacheContent);
+            res = compile_model_impl(model, plugin, _config, context, cacheContent);
         }
```

Comment:
do you think in a separate PR we should merge `load_model_from_cache` and `compile_model_impl` into a single call like:
```c++
load_model_from_cache(cacheContent, plugin, _config, context, []() {
   return compile_model_impl(model, plugin, _config, context, cacheContent)
});
```
So, we will eliminate a check for `loadedFromCache`